app/scripts/metamask-controller.js

```diff
@@ -4266,6 +4272,73 @@ export default class MetamaskController extends EventEmitter {
     }
   }
 
+  /**
+   * Adds a new mnemonic to the vault.
+   *
+   * @param {string} mnemonic
+   * @returns {object} new account address
+   */
+  ///: BEGIN:ONLY_INCLUDE_IF(multi-srp)
+  async addNewMnemonicToVault(mnemonic) {
+    const releaseLock = await this.createVaultMutex.acquire();
+    try {
+      // TODO: remove and use this logic in the keyring controller.
+      const findMostRecentIndex = (arr, targetType) => {
+        return arr.reduceRight((acc, item, index) => {
+          if (acc === -1 && item.type === targetType) {
+            return index;
+          }
+          return acc;
+        }, -1);
+      };
```

Comment:
As discussed here: https://github.com/MetaMask/metamask-extension/pull/29942#discussion_r1965113352

(This could also be moved into some `.ts` file for the time being).

I haven't tested this:

```suggestion
      // TODO: This kind of logic should be inside the `KeyringController` (using `KeyringSelector` query, or make `addNewKeyring` returns it keyring ID alongside
      // its keyring.
      const findKeyringIdByAddress = (address) =>  {
        const { keyrings, keyringsMetadata } = this.keyringController.state;
  
        const keyringIndex = keyrings.findIndex((keyring, index) => {
          return keyring.getAccounts().includes(address);
        });
        if (keyringIndex === -1) {
          throw new Error('Could not find keyring ID, THIS SHOULD NEVER HAPPEN');
        }
        return keyringsMetadata[keyringIndex].id;
      };
```

Reply:
As discussed internally, I think that adding a check with `keyring.type === KeyringTypes.hd` should do the trick?